// @coinbase/cdp-core/dist/web/index.web106.js
import { handleChainId as t } from "./index.web138.js";
import { handleDisconnect as e } from "./index.web139.js";
import { handleSwitchEthereumChain as a } from "./index.web140.js";
import { handleAccounts as n } from "./index.web141.js";
import { handleEthSign as o } from "./index.web142.js";
import { handleGetCallsStatus as r } from "./index.web143.js";
import { handleGetCapabilities as i } from "./index.web144.js";
import { handlePersonalSign as l } from "./index.web145.js";
import { handleRequestAccounts as h } from "./index.web146.js";
import { handleSendCalls as m } from "./index.web147.js";
import { handleSendTransaction as s } from "./index.web148.js";
import { handleSignTypedData as d } from "./index.web149.js";
function E() {
  return {
    eth_accounts: n,
    eth_requestAccounts: h,
    personal_sign: l,
    eth_sendTransaction: s,
    eth_signTypedData_v4: d,
    wallet_disconnect: e,
    wallet_switchEthereumChain: a,
    eth_sign: o,
    eth_chainId: t,
    wallet_sendCalls: m,
    wallet_getCallsStatus: r,
    wallet_getCapabilities: i
  };
}
export {
  E as createSmartAccountHandlers
};
