// @coinbase/cdp-core/dist/web/index.web32.js
import { verifyOAuth as i } from "./index.web6.js";
import { getPlatformOAuth as h } from "./index.web2.js";
import "@coinbase/cdp-api-client";
import "viem";
class A {
  oauthFlowState = null;
  oauthStateChangeCallbacks = [];
  /**
   * Adds a callback to be called when the OAuth state changes.
   *
   * @param callback - The function to call when the OAuth state changes.
   */
  async addOAuthStateChangeCallback(t) {
    this.oauthStateChangeCallbacks.push(t), t(this.oauthFlowState);
  }
  /**
   * Sets the OAuth flow state.
   *
   * @param oauthFlowState - The OAuth flow state.
   */
  async setOAuthFlowState(t) {
    this.oauthFlowState = t, this.oauthStateChangeCallbacks.forEach((a) => a(this.oauthFlowState));
  }
  /**
   * Clears the OAuth flow state.
   */
  async clearOAuthFlowState() {
    this.oauthFlowState = null, this.oauthStateChangeCallbacks.forEach((t) => t(null));
  }
  /**
   * Gets the OAuth flow state.
   *
   * @returns The OAuth flow state.
   */
  async getOAuthFlowState() {
    return this.oauthFlowState;
  }
  /**
   * Handles the OAuth code.
   *
   * @param url - The URL to handle.
   * @returns A promise that resolves when the OAuth code is handled.
   * @throws An error if the URL is required and not provided.
   */
  async handleOAuthCode(t) {
    try {
      const a = await h().getOAuthParams(t);
      if (await h().cleanupOAuthParams(), !a)
        return;
      const { providerType: e, code: r, flowId: o, error: s, errorDescription: u } = a;
      if (s || u) {
        await this.setOAuthFlowState({
          providerType: e,
          status: "error",
          errorDescription: s ?? u,
          code: r,
          flowId: o,
          error: s
        });
        return;
      }
      if (!r || !e || !o)
        return;
      await this.setOAuthFlowState({
        providerType: e,
        status: "pending",
        code: r,
        flowId: o
      }), await i({
        code: r,
        providerType: e,
        flowId: o
      }), await this.setOAuthFlowState({
        providerType: e,
        status: "success",
        code: r,
        flowId: o
      });
    } catch (a) {
      const e = a instanceof Error ? a.message : "Unknown error";
      await h().cleanupOAuthParams(), await this.setOAuthFlowState({
        status: "error",
        errorDescription: e
      });
    }
  }
}
export {
  A as OAuthManager
};
