// @coinbase/cdp-core/dist/web/index.web84.js
import { BaseError as t } from "./index.web76.js";
function o(n) {
  const r = Object.entries(n).map(([a, e]) => e === void 0 || e === !1 ? null : [a, e]).filter(Boolean), i = r.reduce((a, [e]) => Math.max(a, e.length), 0);
  return r.map(([a, e]) => `  ${`${a}:`.padEnd(i + 1)}  ${e}`).join(`
`);
}
class c extends t {
  constructor({ transaction: r }) {
    super("Cannot infer a transaction type from provided transaction.", {
      metaMessages: [
        "Provided Transaction:",
        "{",
        o(r),
        "}",
        "",
        "To infer the type, either provide:",
        "- a `type` to the Transaction, or",
        "- an EIP-1559 Transaction with `maxFeePerGas`, or",
        "- an EIP-2930 Transaction with `gasPrice` & `accessList`, or",
        "- an EIP-4844 Transaction with `blobs`, `blobVersionedHashes`, `sidecars`, or",
        "- an EIP-7702 Transaction with `authorizationList`, or",
        "- a Legacy Transaction with `gasPrice`"
      ],
      name: "InvalidSerializableTransactionError"
    });
  }
}
class d extends t {
  constructor({ storageKey: r }) {
    super(`Size for storage key "${r}" is invalid. Expected 32 bytes. Got ${Math.floor((r.length - 2) / 2)} bytes.`, { name: "InvalidStorageKeySizeError" });
  }
}
export {
  c as InvalidSerializableTransactionError,
  d as InvalidStorageKeySizeError,
  o as prettyPrint
};
