// @coinbase/cdp-core/dist/web/index.web.js
import { setPlatformServices as r } from "./index.web2.js";
import { webCrypto as o } from "./index.web3.js";
import { webOAuth as e } from "./index.web4.js";
import { webStorage as t } from "./index.web5.js";
import { authenticateWithJWT as E, exportEvmAccount as l, exportSolanaAccount as u, getAccessToken as g, getAccessTokenExpiration as x, getCurrentUser as A, getUserOperation as S, initialize as d, isSignedIn as h, onAuthStateChange as v, onOAuthStateChange as T, sendEvmTransaction as O, sendSolanaTransaction as y, sendUserOperation as P, signEvmHash as k, signEvmMessage as W, signEvmTransaction as C, signEvmTypedData as I, signInWithEmail as U, signInWithOAuth as w, signInWithSms as D, signOut as R, signSolanaMessage as b, signSolanaTransaction as M, verifyEmailOTP as N, verifyOAuth as V, verifySmsOTP as B } from "./index.web6.js";
import { createSpendPermission as _ } from "./index.web7.js";
import { listSpendPermissions as G } from "./index.web8.js";
import { APIError as X, ErrorType as j, HttpErrorType as q, OAuth2ProviderType as F, SendEvmTransactionWithEndUserAccountBodyNetwork as K, SendSolanaTransactionWithEndUserAccountBodyNetwork as L } from "@coinbase/cdp-api-client";
import { revokeSpendPermission as Y } from "./index.web9.js";
import { getCurrentUserSync as $ } from "./index.web10.js";
import { fetchWithX402 as or } from "./index.web11.js";
import { linkGoogle as tr } from "./index.web12.js";
import { linkSms as ir } from "./index.web13.js";
import { linkEmail as mr } from "./index.web14.js";
import { linkOAuth as pr } from "./index.web15.js";
import { linkApple as fr } from "./index.web16.js";
import { createEvmEoaAccount as lr } from "./index.web17.js";
import { createEvmSmartAccount as gr } from "./index.web18.js";
import { createSolanaAccount as Ar } from "./index.web19.js";
import { AuthManager as dr } from "./index.web20.js";
import { CustomAuthError as vr } from "./index.web21.js";
import { toViemAccount as Or } from "./index.web22.js";
import { InputValidationError as Pr, InternalError as kr, ModuleResolutionError as Wr, ValidationError as Cr } from "./index.web23.js";
import { createCDPEmbeddedWallet as Ur } from "./index.web24.js";
import { EIP1193ProviderError as Dr, STANDARD_ERROR_CODES as Rr } from "./index.web25.js";
import "viem";
import "zustand";
r({
  crypto: o,
  storage: t,
  oauth: e
});
export {
  X as APIError,
  dr as AuthManager,
  vr as CustomAuthError,
  Dr as EIP1193ProviderError,
  j as ErrorType,
  q as HttpErrorType,
  Pr as InputValidationError,
  kr as InternalError,
  Wr as ModuleResolutionError,
  F as OAuth2ProviderType,
  Rr as STANDARD_ERROR_CODES,
  K as SendEvmTransactionWithEndUserAccountBodyNetwork,
  L as SendSolanaTransactionWithEndUserAccountBodyNetwork,
  Cr as ValidationError,
  E as authenticateWithJWT,
  Ur as createCDPEmbeddedWallet,
  lr as createEvmEoaAccount,
  gr as createEvmSmartAccount,
  Ar as createSolanaAccount,
  _ as createSpendPermission,
  l as exportEvmAccount,
  u as exportSolanaAccount,
  or as fetchWithX402,
  g as getAccessToken,
  x as getAccessTokenExpiration,
  A as getCurrentUser,
  $ as getCurrentUserSync,
  S as getUserOperation,
  d as initialize,
  h as isSignedIn,
  fr as linkApple,
  mr as linkEmail,
  tr as linkGoogle,
  pr as linkOAuth,
  ir as linkSms,
  G as listSpendPermissions,
  v as onAuthStateChange,
  T as onOAuthStateChange,
  Y as revokeSpendPermission,
  O as sendEvmTransaction,
  y as sendSolanaTransaction,
  P as sendUserOperation,
  k as signEvmHash,
  W as signEvmMessage,
  C as signEvmTransaction,
  I as signEvmTypedData,
  U as signInWithEmail,
  w as signInWithOAuth,
  D as signInWithSms,
  R as signOut,
  b as signSolanaMessage,
  M as signSolanaTransaction,
  Or as toViemAccount,
  N as verifyEmailOTP,
  V as verifyOAuth,
  B as verifySmsOTP
};
