// @coinbase/cdp-core/dist/web/index.web4.js
const s = {
  getRedirectUrl: () => (r(), window.location.href),
  initiateOAuthFlow: (o) => {
    r(), window.location.href = o;
  },
  getOAuthParams: () => {
    r();
    const o = new URL(window.location.href), e = new URLSearchParams(o.search), t = e.get("code") ?? void 0, i = e.get("provider_type") ?? void 0, n = e.get("flow_id") ?? void 0, d = e.get("error") ?? void 0, c = e.get("error_description") ?? void 0;
    return (!t || !i || !n) && !d && !c ? null : { code: t, providerType: i, flowId: n, error: d, errorDescription: c };
  },
  cleanupOAuthParams: () => {
    r();
    const o = new URL(window.location.href), e = new URLSearchParams(o.search);
    e.delete("code"), e.delete("provider_type"), e.delete("flow_id"), e.delete("error"), e.delete("error_description"), o.search = e.toString(), window.history.replaceState({}, "", o.toString());
  }
};
function r() {
  if (typeof window > "u" || typeof window.location > "u" || typeof window.history > "u")
    throw new Error("OAuth is not supported in this browser");
}
export {
  s as webOAuth
};
