// @coinbase/cdp-core/dist/web/index.web46.js
import { sliceHex as s, encodePacked as c, encodeAbiParameters as p } from "viem";
import { signEvmTypedData as m } from "./index.web6.js";
async function b(e) {
  const { smartAccount: t, hash: a, chainId: n, ownerAddress: r } = e, o = u({
    hash: a,
    chainId: n,
    smartAccountAddress: t
  }), { signature: i } = await m({
    evmAccount: r,
    typedData: o
  });
  return {
    signature: y({
      signatureHex: i
    })
  };
}
function u({
  hash: e,
  chainId: t,
  smartAccountAddress: a
}) {
  return {
    domain: {
      name: "Coinbase Smart Wallet",
      version: "1",
      chainId: t,
      verifyingContract: a
    },
    types: {
      EIP712Domain: [
        { name: "name", type: "string" },
        { name: "version", type: "string" },
        { name: "chainId", type: "uint256" },
        { name: "verifyingContract", type: "address" }
      ],
      CoinbaseSmartWalletMessage: [{ name: "hash", type: "bytes32" }]
    },
    primaryType: "CoinbaseSmartWalletMessage",
    message: {
      hash: e
    }
  };
}
function y({ signatureHex: e }) {
  const t = s(e, 0, 32), a = s(e, 32, 64), n = +`0x${e.slice(130, 132)}`, r = c(["bytes32", "bytes32", "uint8"], [t, a, n]);
  return p(
    [d],
    [
      {
        ownerIndex: 0,
        signatureData: r
      }
    ]
  );
}
const d = {
  components: [
    {
      name: "ownerIndex",
      type: "uint8"
    },
    {
      name: "signatureData",
      type: "bytes"
    }
  ],
  name: "SignatureWrapper",
  type: "tuple"
};
export {
  u as createReplaySafeTypedData,
  y as createSmartAccountSignatureWrapper,
  b as signAndWrapTypedDataForSmartAccount
};
