// @coinbase/cdp-core/dist/web/index.web143.js
import { getUserOperation as u } from "./index.web6.js";
import { isChainIdSupportedForCDPUserOps as p, userOperationChainIdToNameMapping as r, userOperationNameToChainIdMapping as m } from "./index.web38.js";
import { validateUserHasEvmSmartAccount as d, RPCRequestError as o, STANDARD_ERROR_CODES as n } from "./index.web25.js";
async function I(t, s) {
  const i = t[0], c = (await d()).evmSmartAccounts?.[0], a = s.getState().chainId;
  if (!p(a) || !r[a])
    throw new o(
      n.rpc.invalidParams,
      "Chain ID is not supported for user operations"
    );
  try {
    const e = await u({
      userOperationHash: i,
      evmSmartAccount: c,
      network: r[a]
    });
    return l(e);
  } catch (e) {
    throw console.log("Failed to get user operation status", e), new o(
      n.rpc.internal,
      "Failed to get user operation status"
    );
  }
}
function l(t) {
  return {
    atomic: !0,
    chainId: m[t.network] ?? 0,
    receipts: t.receipts?.map((s) => ({
      transactionHash: s.transactionHash,
      blockHash: s.blockHash,
      blockNumber: BigInt(s.blockNumber ?? 0),
      gasUsed: BigInt(s.gasUsed ?? 0),
      logs: [],
      status: ""
    })),
    id: t.userOpHash,
    status: h(t.status),
    version: "1"
  };
}
function h(t) {
  switch (t) {
    case "dropped":
    case "failed":
      return 400;
    case "complete":
      return 200;
    default:
      return 100;
  }
}
export {
  I as handleGetCallsStatus
};
