// @coinbase/cdp-core/dist/web/index.web7.js
import { createSpendPermissionWithEndUserAccount as a } from "@coinbase/cdp-api-client";
import { Analytics as o } from "./index.web28.js";
import "./index.web29.js";
import { withAuth as i } from "./index.web34.js";
import { toSpendPermission as m } from "./index.web41.js";
import { getConfig as t, getCoreAuthManager as c } from "./index.web39.js";
const h = async (e) => (o.sendActionCallEvent({ name: "create_spend_permission" }), t().useMock ? {
  userOperationHash: "0x1234567890123456789012345678901234567890123456789012345678901234"
} : i(e, c(), async ({ user: n, walletSecretId: s }) => {
  const r = m(
    { ...e, account: e.evmSmartAccount },
    e.network
  );
  return {
    userOperationHash: (await a(
      t().projectId,
      n.userId,
      e.evmSmartAccount,
      {
        // spend permission options
        spender: r.spender,
        token: r.token,
        allowance: r.allowance.toString(),
        period: r.period.toString(),
        start: r.start.toString(),
        end: r.end.toString(),
        salt: r.salt.toString(),
        // api options
        walletSecretId: s,
        network: e.network,
        useCdpPaymaster: e.useCdpPaymaster ?? !1,
        paymasterUrl: e.paymasterUrl
      }
    )).userOpHash
  };
}));
export {
  h as createSpendPermission
};
