// @coinbase/cdp-core/dist/esm/index.web136.js
import { createWalletClient as u } from "viem";
import { sendEvmTransaction as l, signEvmTransaction as w } from "./index.web6.js";
import { isChainIdSupportedForCDPSends as v, sendTransactionChainIdToNameMapping as T } from "./index.web38.js";
import { RPCRequestError as t, STANDARD_ERROR_CODES as i, validateUserOwnsEOA as P } from "./index.web25.js";
async function E(e, d, m) {
  const [a] = e;
  if (!a || typeof a != "object")
    throw new t(
      i.rpc.invalidParams,
      "Transaction parameter must be an object"
    );
  if (!a.to)
    throw new t(
      i.rpc.invalidParams,
      "Transaction must include 'to' field"
    );
  await P(a.from);
  const o = d.getState(), r = a.chainId ? Number(a.chainId) : o.chainId;
  if (v(r))
    try {
      const { transactionHash: n } = await l({
        evmAccount: a.from,
        transaction: { ...c(a), chainId: r },
        network: T[r]
      });
      return n;
    } catch (n) {
      throw new t(
        i.rpc.transactionRejected,
        n instanceof Error ? n.message : "Transaction failed"
      );
    }
  else
    try {
      const n = u({
        // Safe as we check before calling this handler that chainId is configured
        chain: o.chains.find((s) => s.id === r),
        transport: m[r],
        account: a.from
      }), p = o.chains.find((s) => s.id === r), h = await n.prepareTransactionRequest({
        ...c(a),
        chain: p
      }), { signedTransaction: f } = await w({
        evmAccount: a.from,
        transaction: {
          ...h,
          type: "eip1559"
        }
      });
      return await n.sendRawTransaction({
        serializedTransaction: f
      });
    } catch (n) {
      throw new t(
        i.rpc.transactionRejected,
        n instanceof Error ? n.message : "Transaction failed"
      );
    }
}
function c(e) {
  return {
    to: e.to,
    data: e.data,
    value: e.value ? BigInt(e.value) : void 0,
    nonce: e.nonce ? Number(e.nonce) : void 0,
    gas: e.gas ? BigInt(e.gas) : void 0,
    maxFeePerGas: e.maxFeePerGas ? BigInt(e.maxFeePerGas) : void 0,
    maxPriorityFeePerGas: e.maxPriorityFeePerGas ? BigInt(e.maxPriorityFeePerGas) : void 0,
    type: "eip1559"
  };
}
export {
  E as handleSendTransaction
};
