// @coinbase/cdp-core/dist/esm/index.native59.js
import { encoder as n } from "./index.native109.js";
import e from "./index.native110.js";
import i from "./index.native111.js";
import o from "./index.native112.js";
function s(r, t) {
  if (!Number.isFinite(t))
    throw new TypeError(`Invalid ${r} input`);
  return t;
}
class d {
  #t;
  constructor(t) {
    if (!o(t))
      throw new TypeError("JWT Claims Set MUST be an object");
    this.#t = structuredClone(t);
  }
  data() {
    return n.encode(JSON.stringify(this.#t));
  }
  get iss() {
    return this.#t.iss;
  }
  set iss(t) {
    this.#t.iss = t;
  }
  get sub() {
    return this.#t.sub;
  }
  set sub(t) {
    this.#t.sub = t;
  }
  get aud() {
    return this.#t.aud;
  }
  set aud(t) {
    this.#t.aud = t;
  }
  set jti(t) {
    this.#t.jti = t;
  }
  set nbf(t) {
    typeof t == "number" ? this.#t.nbf = s("setNotBefore", t) : t instanceof Date ? this.#t.nbf = s("setNotBefore", e(t)) : this.#t.nbf = e(/* @__PURE__ */ new Date()) + i(t);
  }
  set exp(t) {
    typeof t == "number" ? this.#t.exp = s("setExpirationTime", t) : t instanceof Date ? this.#t.exp = s("setExpirationTime", e(t)) : this.#t.exp = e(/* @__PURE__ */ new Date()) + i(t);
  }
  set iat(t) {
    typeof t > "u" ? this.#t.iat = e(/* @__PURE__ */ new Date()) : t instanceof Date ? this.#t.iat = s("setIssuedAt", e(t)) : typeof t == "string" ? this.#t.iat = s("setIssuedAt", e(/* @__PURE__ */ new Date()) + i(t)) : this.#t.iat = s("setIssuedAt", t);
  }
}
export {
  d as JWTClaimsBuilder
};
