// @coinbase/cdp-core/dist/esm/index.native24.js
class n extends Error {
  /**
   * The name of the missing module.
   */
  moduleName;
  /**
   * The API or feature that requires the missing module.
   */
  requiredFor;
  /**
   * Creates a new ModuleResolutionError.
   *
   * @param moduleName - The name of the missing module.
   * @param requiredFor - The API or feature that requires the module.
   * @param message - Optional custom error message.
   */
  constructor(r, s, t) {
    const o = `${s} is not available in this React Native environment. Please ensure ${r} is installed.`;
    super(t || o), this.name = "ModuleResolutionError", this.moduleName = r, this.requiredFor = s;
  }
}
class a extends Error {
  /**
   * Creates a new InputValidationError.
   *
   * @param message - The validation error message.
   */
  constructor(r) {
    super(r), this.name = "InputValidationError";
  }
}
class i extends Error {
  /**
   * Creates a new ValidationError.
   *
   * @param message - The validation error message.
   */
  constructor(r) {
    super(r), this.name = "ValidationError";
  }
}
class l extends Error {
  /**
   * Creates a new InternalError.
   *
   * @param message - The internal error message.
   */
  constructor(r) {
    super(r), this.name = "InternalError";
  }
}
export {
  a as InputValidationError,
  l as InternalError,
  n as ModuleResolutionError,
  i as ValidationError
};
