// @coinbase/cdp-core/dist/esm/index.native30.js
import p from "./index.native137.js";
import { getFramework as m } from "./index.native138.js";
const y = "1f8d3e558f493e38a037dbfadb8ba344", v = "d473b1dcc969105adab157226cd9cf63", i = Symbol.for("@coinbase/cdp-analytics");
function g() {
  const t = globalThis;
  return t[i] || (t[i] = {
    identifier: "",
    enabled: !0,
    isProduction: !1,
    isUsingMocks: !1,
    sendEvent: o,
    sendActionCallEvent: (e) => o({ ...e, type: "action_call" }),
    sendHookCallEvent: (e) => o({ ...e, type: "hook_call" }),
    sendComponentCallEvent: (e) => o({ ...e, type: "component_call" }),
    versionRegistry: {},
    registerPackageVersion: (e, s) => {
      n.versionRegistry[e] = s;
    }
  }), t[i];
}
const n = g();
async function o(t) {
  if (!n.enabled || !n.identifier)
    return;
  const e = Date.now(), r = [{
    user_id: n.identifier,
    event_type: t.type,
    platform: c() ? "web" : "native",
    timestamp: e,
    event_properties: {
      project_name: "cdp-embedded-wallet",
      framework: m(),
      isLocalHost: u(),
      isUsingMocks: n.isUsingMocks,
      package_versions: n.versionRegistry,
      ...t
    }
  }], a = JSON.stringify(r), d = e.toString(), l = p(a + d), f = {
    client: n.isProduction ? v : y,
    e: a,
    checksum: l
  };
  await fetch("https://cca-lite.coinbase.com/amp", {
    method: "POST",
    mode: "no-cors",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(f)
  });
}
function c() {
  return typeof window < "u" && typeof document < "u";
}
function u() {
  return c() ? window.location.hostname === "localhost" || window.location.hostname === "127.0.0.1" : !1;
}
export {
  n as Analytics
};
