// @coinbase/cdp-core/dist/esm/index.web105.js
import { handleAccounts as t } from "./index.web129.js";
import { handleEthSign as e } from "./index.web130.js";
import { handleGetCallsStatus as a } from "./index.web131.js";
import { handleGetCapabilities as n } from "./index.web132.js";
import { handlePersonalSign as o } from "./index.web133.js";
import { handleRequestAccounts as r } from "./index.web134.js";
import { handleSendCalls as i } from "./index.web135.js";
import { handleSendTransaction as l } from "./index.web136.js";
import { handleSignTypedData as h } from "./index.web137.js";
import { handleChainId as s } from "./index.web138.js";
import { handleDisconnect as m } from "./index.web139.js";
import { handleSwitchEthereumChain as d } from "./index.web140.js";
function D() {
  return {
    eth_accounts: t,
    eth_requestAccounts: r,
    personal_sign: o,
    eth_sendTransaction: l,
    eth_signTypedData_v4: h,
    wallet_disconnect: m,
    wallet_switchEthereumChain: d,
    eth_sign: e,
    eth_chainId: s,
    wallet_sendCalls: i,
    wallet_getCallsStatus: a,
    wallet_getCapabilities: n
  };
}
export {
  D as createEOAHandlers
};
