// @coinbase/cdp-core/dist/esm/index.web42.js
import { getTransactionEncoder as u, getTransactionDecoder as f, address as g } from "@solana/kit";
import { getCurrentUserSync as l } from "./index.web10.js";
import { signSolanaTransaction as m } from "./index.web6.js";
import { encodeBase64 as T, decodeBase64 as h } from "./index.web26.js";
function E(n) {
  const o = l();
  if (!o || !o.solanaAccounts?.length)
    throw new Error(
      "No accounts found. User must be authenticated and have at least one Solana account"
    );
  const r = u(), a = f();
  return {
    address: g(n),
    signTransactions: async (c) => {
      const t = [];
      for (const s of c) {
        const i = new Uint8Array(r.encode(s)), { signedTransaction: d } = await m({
          solanaAccount: n,
          transaction: T(i)
        }), e = a.decode(h(d)).signatures[n];
        if (!e)
          throw new Error("Wallet did not return a signature for the selected account.");
        t.push(
          Object.freeze({
            [n]: e
          })
        );
      }
      return t;
    }
  };
}
export {
  E as toTransactionSigner
};
