// @coinbase/cdp-core/dist/esm/index.native45.js
import { CompactSign as i } from "./index.native57.js";
import { JWTInvalid as r } from "./index.native58.js";
import { JWTClaimsBuilder as n } from "./index.native59.js";
class d {
  #s;
  #t;
  constructor(t = {}) {
    this.#t = new n(t);
  }
  setIssuer(t) {
    return this.#t.iss = t, this;
  }
  setSubject(t) {
    return this.#t.sub = t, this;
  }
  setAudience(t) {
    return this.#t.aud = t, this;
  }
  setJti(t) {
    return this.#t.jti = t, this;
  }
  setNotBefore(t) {
    return this.#t.nbf = t, this;
  }
  setExpirationTime(t) {
    return this.#t.exp = t, this;
  }
  setIssuedAt(t) {
    return this.#t.iat = t, this;
  }
  setProtectedHeader(t) {
    return this.#s = t, this;
  }
  async sign(t, e) {
    const s = new i(this.#t.data());
    if (s.setProtectedHeader(this.#s), Array.isArray(this.#s?.crit) && this.#s.crit.includes("b64") && this.#s.b64 === !1)
      throw new r("JWTs MUST NOT use unencoded payload");
    return s.sign(t, e);
  }
}
export {
  d as SignJWT
};
