// @coinbase/cdp-core/dist/esm/index.native38.js
import "@coinbase/cdp-api-client";
import "viem";
import { verifyOAuth as o } from "./index.native7.js";
import { getPlatformOAuth as e } from "./index.native6.js";
class i {
  oauthFlowState = null;
  oauthStateChangeCallbacks = [];
  /**
   * Initializes the mock OAuth manager.
   */
  constructor() {
    this.handleOAuthCode();
  }
  /**
   * Gets the current user, or null if there is no user signed in.
   *
   * @returns The current mock user if signed in, null otherwise.
   */
  async getOAuthFlowState() {
    return this.oauthFlowState ?? null;
  }
  /**
   * Sets the OAuth flow state.
   *
   * @param oauthFlowState - The OAuth flow state to set.
   */
  async setOAuthFlowState(t) {
    this.oauthFlowState = t, this.oauthStateChangeCallbacks.forEach((a) => a(this.oauthFlowState));
  }
  /**
   * Adds a callback to be called when the OAuth state changes.
   *
   * @param callback - The function to call when the OAuth state changes.
   */
  async addOAuthStateChangeCallback(t) {
    this.oauthStateChangeCallbacks.push(t), t(this.oauthFlowState);
  }
  /**
   * Clears the OAuth flow state.
   */
  async clearOAuthFlowState() {
    this.oauthFlowState = null, this.oauthStateChangeCallbacks.forEach((t) => t(null));
  }
  /**
   * Awaitable method whose promise only resolves when the OAuth manager is ready to be used.
   */
  async handleOAuthCode() {
    await e()?.cleanupOAuthParams(), await this.setOAuthFlowState({
      providerType: "google",
      status: "pending",
      code: "mock-oauth-code",
      flowId: "mock-flow-id"
    }), await o({ code: "mock-oauth-code", providerType: "google", flowId: "mock-flow-id" }), await this.setOAuthFlowState({
      providerType: "google",
      status: "success",
      code: "mock-oauth-code",
      flowId: "mock-flow-id"
    });
  }
}
export {
  i as MockOAuthManager
};
