// @coinbase/cdp-core/dist/esm/index.web3.js
import { decodeBase64 as y, encodeBase64 as a } from "./index.web26.js";
import "@coinbase/cdp-api-client";
import "viem";
const p = {
  /**
   * Creates a secp256r1 key pair for the browser.
   *
   * @returns A secp256r1 key pair with private key and base64-encoded public key.
   */
  async createKeyPair() {
    const e = await window.crypto.subtle.generateKey(
      {
        name: "ECDSA",
        namedCurve: "P-256"
        // secp256r1
      },
      !1,
      // Do not allow key export.
      ["sign", "verify"]
      // Key usages
    ), t = await window.crypto.subtle.exportKey("spki", e.publicKey), r = new Uint8Array(t), n = a(r);
    return { privateKey: e.privateKey, publicKeyBase64: n };
  },
  /**
   * Creates an RSA key pair that allows private key export.
   *
   * @returns An RSA key pair with exportable private key and base64-encoded public key.
   */
  async createExportKeyPair() {
    const e = await window.crypto.subtle.generateKey(
      {
        name: "RSA-OAEP",
        modulusLength: 2048,
        publicExponent: new Uint8Array([1, 0, 1]),
        hash: "SHA-256"
      },
      !0,
      // Allow key export
      ["encrypt", "decrypt"]
    ), t = await window.crypto.subtle.exportKey("spki", e.publicKey), r = a(new Uint8Array(t));
    return { privateKey: e.privateKey, publicKeyBase64: r };
  },
  /**
   * Computes a hash of the given data.
   *
   * @param data - The data to hash.
   * @returns Promise that resolves to the hex-encoded hash.
   */
  async hash(e) {
    const t = await window.crypto.subtle.digest("SHA-256", e), r = new Uint8Array(t);
    return Array.from(r).map((o) => o.toString(16).padStart(2, "0")).join("");
  },
  /**
   * Generates a cryptographically secure random identifier.
   *
   * @returns A random string suitable for use as an identifier.
   */
  generateRandomId() {
    const e = new Uint8Array(16);
    return window.crypto.getRandomValues(e), Array.from(e, (t) => t.toString(16).padStart(2, "0")).join("");
  },
  /**
   * Generates a cryptographically secure random salt.
   *
   * @returns A random bigint salt.
   */
  generateRandomSalt() {
    const e = new Uint8Array(32);
    window.crypto.getRandomValues(e);
    let t = 0n;
    for (let r = 0; r < e.length; r++)
      t = (t << 8n) + BigInt(e[r]);
    return t;
  },
  /**
   * Decrypts data using a private key with RSA-OAEP algorithm.
   *
   * @param privateKey - The private key to use for decryption.
   * @param encryptedData - Base64-encoded encrypted data.
   * @returns Promise that resolves to the decrypted string.
   */
  async decryptWithPrivateKey(e, t) {
    const r = y(t);
    return await window.crypto.subtle.decrypt(
      "RSA-OAEP",
      e,
      r
    );
  }
};
export {
  p as webCrypto
};
