// @coinbase/cdp-core/dist/esm/index.web49.js
class n {
  events = {};
  /**
   * Add an event listener.
   *
   * @param event - The name of the event to listen for.
   * @param listener - The function to call when the event is emitted.
   */
  on(e, s) {
    this.events[e] || (this.events[e] = []), this.events[e].push(s);
  }
  /**
   * Remove an event listener.
   *
   * @param event - The name of the event to remove the listener from.
   * @param listener - The function to remove from the event listeners.
   */
  removeListener(e, s) {
    if (!this.events[e])
      return;
    const t = this.events[e].indexOf(s);
    t > -1 && this.events[e].splice(t, 1);
  }
  /**
   * Emit an event.
   *
   * @param event - The name of the event to emit.
   * @param args - The arguments to pass to the event listeners.
   */
  emit(e, ...s) {
    this.events[e] && this.events[e].forEach((t) => {
      try {
        t(...s);
      } catch (i) {
        console.error(`Error in event listener for ${e}:`, i);
      }
    });
  }
  /**
   * Remove all listeners for an event.
   *
   * @param event - The name of the event to remove all listeners from. If not provided, removes all listeners for all events.
   */
  removeAllListeners(e) {
    e ? delete this.events[e] : this.events = {};
  }
}
export {
  n as EventEmitter
};
