// @coinbase/cdp-core/dist/esm/index.web91.js
import { formatGwei as a } from "./index.web97.js";
import { BaseError as o } from "./index.web76.js";
class n extends o {
  constructor({ cause: r, maxFeePerGas: e } = {}) {
    super(`The fee cap (\`maxFeePerGas\`${e ? ` = ${a(e)} gwei` : ""}) cannot be higher than the maximum allowed value (2^256-1).`, {
      cause: r,
      name: "FeeCapTooHighError"
    });
  }
}
Object.defineProperty(n, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /max fee per gas higher than 2\^256-1|fee cap higher than 2\^256-1/
});
class h extends o {
  constructor({ cause: r, maxPriorityFeePerGas: e, maxFeePerGas: t } = {}) {
    super([
      `The provided tip (\`maxPriorityFeePerGas\`${e ? ` = ${a(e)} gwei` : ""}) cannot be higher than the fee cap (\`maxFeePerGas\`${t ? ` = ${a(t)} gwei` : ""}).`
    ].join(`
`), {
      cause: r,
      name: "TipAboveFeeCapError"
    });
  }
}
Object.defineProperty(h, "nodeMessage", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: /max priority fee per gas higher than max fee per gas|tip higher than fee cap/
});
export {
  n as FeeCapTooHighError,
  h as TipAboveFeeCapError
};
