// @coinbase/cdp-core/dist/esm/index.native66.js
function O(a, e = {}) {
  const { raw: r = !1 } = e, o = a;
  if (r)
    return a;
  if (o.error)
    throw P(o.error);
  return o.result;
}
function P(a) {
  const e = a;
  if (e instanceof Error && !("code" in e))
    return new u({
      cause: e,
      data: e,
      message: e.message
    });
  const { code: r } = e;
  return r === u.code ? new u(e) : r === n.code ? new n(e) : r === m.code ? new m(e) : r === f.code ? new f(e) : r === d.code ? new d(e) : r === p.code ? new p(e) : r === l.code ? new l(e) : r === g.code ? new g(e) : r === c.code ? new c(e) : r === s.code ? new s(e) : r === i.code ? new i(e) : r === b.code ? new b(e) : new u({
    cause: e instanceof Error ? e : void 0,
    data: e,
    message: e.message
  });
}
class t extends Error {
  constructor(e) {
    const { cause: r, code: o, message: w, data: v } = e;
    super(w, { cause: r }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.BaseError"
    }), Object.defineProperty(this, "cause", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "data", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.cause = r, this.code = o, this.data = v;
  }
}
class n extends t {
  constructor(e = {}) {
    super({
      code: n.code,
      data: e.data,
      message: e.message ?? "Missing or invalid parameters."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32e3
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.InvalidInputError"
    });
  }
}
Object.defineProperty(n, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32e3
});
class c extends t {
  constructor(e = {}) {
    super({
      code: c.code,
      data: e.data,
      message: e.message ?? "Requested resource not found."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32001
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.ResourceNotFoundError"
    });
  }
}
Object.defineProperty(c, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32001
});
class s extends t {
  constructor(e = {}) {
    super({
      code: s.code,
      data: e.data,
      message: e.message ?? "Requested resource not available."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32002
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.ResourceUnavailableError"
    });
  }
}
Object.defineProperty(s, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32002
});
class i extends t {
  constructor(e = {}) {
    super({
      code: i.code,
      data: e.data,
      message: e.message ?? "Transaction creation failed."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32003
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.TransactionRejectedError"
    });
  }
}
Object.defineProperty(i, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32003
});
class l extends t {
  constructor(e = {}) {
    super({
      code: l.code,
      data: e.data,
      message: e.message ?? "Method is not implemented."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32004
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.MethodNotSupportedError"
    });
  }
}
Object.defineProperty(l, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32004
});
class d extends t {
  constructor(e = {}) {
    super({
      code: d.code,
      data: e.data,
      message: e.message ?? "Rate limit exceeded."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32005
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.LimitExceededError"
    });
  }
}
Object.defineProperty(d, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32005
});
class b extends t {
  constructor(e = {}) {
    super({
      code: b.code,
      data: e.data,
      message: e.message ?? "JSON-RPC version not supported."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32006
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.VersionNotSupportedError"
    });
  }
}
Object.defineProperty(b, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32006
});
class f extends t {
  constructor(e = {}) {
    super({
      code: f.code,
      data: e.data,
      message: e.message ?? "Input is not a valid JSON-RPC request."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32600
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.InvalidRequestError"
    });
  }
}
Object.defineProperty(f, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32600
});
class p extends t {
  constructor(e = {}) {
    super({
      code: p.code,
      data: e.data,
      message: e.message ?? "Method does not exist."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32601
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.MethodNotFoundError"
    });
  }
}
Object.defineProperty(p, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32601
});
class m extends t {
  constructor(e = {}) {
    super({
      code: m.code,
      data: e.data,
      message: e.message ?? "Invalid method parameters."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32602
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.InvalidParamsError"
    });
  }
}
Object.defineProperty(m, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32602
});
class u extends t {
  constructor(e = {}) {
    super({
      cause: e.cause,
      code: u.code,
      data: e.data,
      message: e.message ?? "Internal JSON-RPC error."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32603
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.InternalError"
    });
  }
}
Object.defineProperty(u, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32603
});
class g extends t {
  constructor(e = {}) {
    super({
      code: g.code,
      data: e.data,
      message: e.message ?? "Failed to parse JSON-RPC response."
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: -32700
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "RpcResponse.ParseError"
    });
  }
}
Object.defineProperty(g, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: -32700
});
export {
  t as BaseError,
  u as InternalError,
  n as InvalidInputError,
  m as InvalidParamsError,
  f as InvalidRequestError,
  d as LimitExceededError,
  p as MethodNotFoundError,
  l as MethodNotSupportedError,
  g as ParseError,
  c as ResourceNotFoundError,
  s as ResourceUnavailableError,
  i as TransactionRejectedError,
  b as VersionNotSupportedError,
  O as parse,
  P as parseError
};
