// @coinbase/cdp-core/dist/esm/index.native63.js
import { handleChainId as t } from "./index.native123.js";
import { handleDisconnect as e } from "./index.native124.js";
import { handleSwitchEthereumChain as a } from "./index.native125.js";
import { handleAccounts as n } from "./index.native126.js";
import { handleEthSign as o } from "./index.native127.js";
import { handleGetCallsStatus as r } from "./index.native128.js";
import { handleGetCapabilities as i } from "./index.native129.js";
import { handlePersonalSign as l } from "./index.native130.js";
import { handleRequestAccounts as h } from "./index.native131.js";
import { handleSendCalls as m } from "./index.native132.js";
import { handleSendTransaction as s } from "./index.native133.js";
import { handleSignTypedData as d } from "./index.native134.js";
function E() {
  return {
    eth_accounts: n,
    eth_requestAccounts: h,
    personal_sign: l,
    eth_sendTransaction: s,
    eth_signTypedData_v4: d,
    wallet_disconnect: e,
    wallet_switchEthereumChain: a,
    eth_sign: o,
    eth_chainId: t,
    wallet_sendCalls: m,
    wallet_getCallsStatus: r,
    wallet_getCapabilities: i
  };
}
export {
  E as createSmartAccountHandlers
};
