// @coinbase/cdp-core/dist/esm/index.native108.js
import { encode as d } from "./index.native144.js";
import y from "./index.native145.js";
import b from "./index.native146.js";
import { JWSInvalid as r } from "./index.native58.js";
import { encoder as s, concat as u, decoder as h } from "./index.native109.js";
import g from "./index.native147.js";
import H from "./index.native148.js";
import P from "./index.native149.js";
class x {
  #r;
  #e;
  #t;
  constructor(e) {
    if (!(e instanceof Uint8Array))
      throw new TypeError("payload must be an instance of Uint8Array");
    this.#r = e;
  }
  setProtectedHeader(e) {
    if (this.#e)
      throw new TypeError("setProtectedHeader can only be called once");
    return this.#e = e, this;
  }
  setUnprotectedHeader(e) {
    if (this.#t)
      throw new TypeError("setUnprotectedHeader can only be called once");
    return this.#t = e, this;
  }
  async sign(e, l) {
    if (!this.#e && !this.#t)
      throw new r("either setProtectedHeader or setUnprotectedHeader must be called before #sign()");
    if (!b(this.#e, this.#t))
      throw new r("JWS Protected and JWS Unprotected Header Parameter names must be disjoint");
    const c = {
      ...this.#e,
      ...this.#t
    }, p = H(r, /* @__PURE__ */ new Map([["b64", !0]]), l?.crit, this.#e, c);
    let o = !0;
    if (p.has("b64") && (o = this.#e.b64, typeof o != "boolean"))
      throw new r('The "b64" (base64url-encode payload) Header Parameter must be a boolean');
    const { alg: t } = c;
    if (typeof t != "string" || !t)
      throw new r('JWS "alg" (Algorithm) Header Parameter missing or invalid');
    g(t, e, "sign");
    let i = this.#r;
    o && (i = s.encode(d(i)));
    let n;
    this.#e ? n = s.encode(d(JSON.stringify(this.#e))) : n = s.encode("");
    const f = u(n, s.encode("."), i), m = await P(e, t), w = await y(t, m, f), a = {
      signature: d(w),
      payload: ""
    };
    return o && (a.payload = h.decode(i)), this.#t && (a.header = this.#t), this.#e && (a.protected = h.decode(n)), a;
  }
}
export {
  x as FlattenedSign
};
