// @coinbase/cdp-core/dist/esm/index.native37.js
import "@coinbase/cdp-api-client";
import "viem";
import { mockUser as e } from "./index.native39.js";
class n {
  authState = null;
  authStateChangeCallback = null;
  /**
   * Initializes the mock auth manager.
   *
   * @param _projectId - The project ID (unused in mock implementation).
   */
  constructor(t) {
  }
  /**
   * Awaitable method whose promise only resolves when the auth manager is ready to be used.
   *
   * @returns A promise that resolves when the auth manager is ready to be used.
   */
  async ensureInitialized() {
    return Promise.resolve();
  }
  /**
   * Gets the current user, or null if there is no user signed in.
   *
   * @returns The current mock user if signed in, null otherwise.
   */
  getUser() {
    return this.authState?.user ?? null;
  }
  /**
   * Returns whether the user is signed in - i.e., whether there is an unexpired
   * access token and user.
   *
   * @returns True if the mock user is signed in and token is not expired, false otherwise.
   */
  async isSignedIn() {
    return this.authState ? Promise.resolve(this.authState.expiresAt > Date.now()) : Promise.resolve(!1);
  }
  /**
   * Signs out the user, clearing all authentication state.
   * In the mock implementation, this simply clears the local auth state.
   */
  async signOut() {
    await this.clearAuthState();
  }
  /**
   * Sets a callback to be called when the auth state changes.
   * The callback is immediately called with the current auth state.
   *
   * @param callback - The function to call when the auth state changes.
   *                  Will be called with the current user or null.
   */
  addAuthStateChangeCallback(t) {
    this.authStateChangeCallback = t, t(this.authState?.user ?? null);
  }
  /**
   * Gets the access token. For mock, this always returns a valid token if signed in.
   * If the current token is expired, it will be refreshed automatically.
   *
   * @returns The mock access token if signed in, null otherwise.
   */
  async getToken() {
    return this.authState ? (this.authState.expiresAt <= Date.now() && await this.refreshToken(), this.authState?.accessToken ?? null) : null;
  }
  /**
   * Gets the expiration time of the access token, or null if the user is not signed in.
   *
   * @returns The expiration time of the access token.
   */
  async getTokenExpiration() {
    return this.authState?.expiresAt ?? null;
  }
  /**
   * Gets the currentlly registered wallet secret ID.
   *
   * @returns The wallet secret ID.
   */
  async getWalletSecretId() {
    return "mock-wallet-secret-id";
  }
  /**
   * Gets the X-Wallet-Auth header value.
   *
   * @param _options - The options for the request.
   * @param _options.requestMethod - The HTTP method of the request.
   * @param _options.requestHost - The host of the request.
   * @param _options.requestPath - The path of the request.
   * @param _options.requestData - The data of the request.
   * @returns The X-Wallet-Auth header value.
   */
  async getXWalletAuth(t) {
    if (!this.authState)
      throw new Error("User not signed in");
    return "x-wallet-auth";
  }
  /**
   * Gets the authentication state.
   *
   * @returns The authentication state.
   */
  getAuthState() {
    return this.authState;
  }
  /**
   * Sets the authentication state.
   * This will trigger the auth state change callback if one is registered.
   *
   * @param authState - The new authentication state to set.
   */
  async setAuthState(t) {
    this.authState = t, this.authStateChangeCallback && this.authStateChangeCallback(this.authState?.user ?? null);
  }
  /**
   * Clears the authentication state.
   * This will trigger the auth state change callback if one is registered.
   */
  async clearAuthState() {
    this.authState = null, this.authStateChangeCallback && this.authStateChangeCallback(null);
  }
  /**
   * Refreshes the mock token.
   * Only refreshes if there is an existing auth state (user is signed in).
   * Sets a new mock token with a 24-hour expiration.
   *
   * @private
   */
  async refreshToken() {
    this.authState && await this.setAuthState({
      accessToken: "mock-access-token",
      expiresAt: Date.now() + 1e3 * 60 * 60 * 24,
      user: e
    });
  }
}
export {
  n as MockAuthManager
};
