// @coinbase/cdp-core/dist/esm/index.native61.js
import "./index.native64.js";
import { BaseError as P } from "./index.native65.js";
import { parse as y, parseError as j, InternalError as g } from "./index.native66.js";
class u extends Error {
  constructor(e, t) {
    super(t), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "ProviderRpcError"
    }), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "details", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.code = e, this.details = t;
  }
}
class n extends u {
  constructor({ message: e = "The user rejected the request." } = {}) {
    super(4001, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4001
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UserRejectedRequestError"
    });
  }
}
Object.defineProperty(n, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4001
});
class a extends u {
  constructor({ message: e = "The requested method and/or account has not been authorized by the user." } = {}) {
    super(4100, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4100
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UnauthorizedError"
    });
  }
}
Object.defineProperty(a, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4100
});
class i extends u {
  constructor({ message: e = "The provider does not support the requested method." } = {}) {
    super(4200, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4200
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UnsupportedMethodError"
    });
  }
}
Object.defineProperty(i, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4200
});
class c extends u {
  constructor({ message: e = "The provider is disconnected from all chains." } = {}) {
    super(4900, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4900
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.DisconnectedError"
    });
  }
}
Object.defineProperty(c, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4900
});
class l extends u {
  constructor({ message: e = "The provider is not connected to the requested chain." } = {}) {
    super(4901, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4901
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.ChainDisconnectedError"
    });
  }
}
Object.defineProperty(l, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4901
});
class d extends u {
  constructor({ message: e = "An error occurred when attempting to switch chain." } = {}) {
    super(4902, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 4902
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.SwitchChainError"
    });
  }
}
Object.defineProperty(d, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 4902
});
class b extends u {
  constructor({ message: e = "This Wallet does not support a capability that was not marked as optional." } = {}) {
    super(5700, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5700
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UnsupportedNonOptionalCapabilityError"
    });
  }
}
Object.defineProperty(b, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5700
});
class s extends u {
  constructor({ message: e = "This Wallet does not support the requested chain ID." } = {}) {
    super(5710, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5710
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UnsupportedChainIdError"
    });
  }
}
Object.defineProperty(s, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5710
});
class f extends u {
  constructor({ message: e = "There is already a bundle submitted with this ID." } = {}) {
    super(5720, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5720
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.DuplicateIdError"
    });
  }
}
Object.defineProperty(f, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5720
});
class p extends u {
  constructor({ message: e = "This bundle id is unknown / has not been submitted." } = {}) {
    super(5730, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5730
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.UnknownBundleIdError"
    });
  }
}
Object.defineProperty(p, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5730
});
class h extends u {
  constructor({ message: e = "The call bundle is too large for the Wallet to process." } = {}) {
    super(5740, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5740
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.BundleTooLargeError"
    });
  }
}
Object.defineProperty(h, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5740
});
class m extends u {
  constructor({ message: e = "The Wallet can support atomicity after an upgrade, but the user rejected the upgrade." } = {}) {
    super(5750, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5750
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.AtomicReadyWalletRejectedUpgradeError"
    });
  }
}
Object.defineProperty(m, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5750
});
class w extends u {
  constructor({ message: e = "The wallet does not support atomic execution but the request requires it." } = {}) {
    super(5760, e), Object.defineProperty(this, "code", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: 5760
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.AtomicityNotSupportedError"
    });
  }
}
Object.defineProperty(w, "code", {
  enumerable: !0,
  configurable: !0,
  writable: !0,
  value: 5760
});
function q(r, e = {}) {
  const { includeEvents: t = !0 } = e;
  if (!r)
    throw new E();
  return {
    ...t ? {
      on: r.on?.bind(r),
      removeListener: r.removeListener?.bind(r)
    } : {},
    async request(v) {
      try {
        const o = await r.request(v);
        return o && typeof o == "object" && "jsonrpc" in o ? y(o) : o;
      } catch (o) {
        throw O(o);
      }
    }
  };
}
function O(r) {
  const e = j(r);
  if (e instanceof g) {
    if (!e.data)
      return e;
    const { code: t } = e.data;
    if (t === c.code)
      return new c(e);
    if (t === l.code)
      return new l(e);
    if (t === n.code)
      return new n(e);
    if (t === a.code)
      return new a(e);
    if (t === i.code)
      return new i(e);
    if (t === d.code)
      return new d(e);
    if (t === m.code)
      return new m(e);
    if (t === w.code)
      return new w(e);
    if (t === h.code)
      return new h(e);
    if (t === p.code)
      return new p(e);
    if (t === f.code)
      return new f(e);
    if (t === s.code)
      return new s(e);
    if (t === b.code)
      return new b(e);
  }
  return e;
}
class E extends P {
  constructor() {
    super("`provider` is undefined."), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Provider.IsUndefinedError"
    });
  }
}
export {
  m as AtomicReadyWalletRejectedUpgradeError,
  w as AtomicityNotSupportedError,
  h as BundleTooLargeError,
  l as ChainDisconnectedError,
  c as DisconnectedError,
  f as DuplicateIdError,
  E as IsUndefinedError,
  u as ProviderRpcError,
  d as SwitchChainError,
  a as UnauthorizedError,
  p as UnknownBundleIdError,
  s as UnsupportedChainIdError,
  i as UnsupportedMethodError,
  b as UnsupportedNonOptionalCapabilityError,
  n as UserRejectedRequestError,
  q as from,
  O as parseError
};
