// @coinbase/cdp-core/dist/esm/index.web102.js
import { getVersion as n } from "./index.web121.js";
class s extends Error {
  constructor(r, e = {}) {
    const a = (() => {
      if (e.cause instanceof s) {
        if (e.cause.details)
          return e.cause.details;
        if (e.cause.shortMessage)
          return e.cause.shortMessage;
      }
      return e.cause && "details" in e.cause && typeof e.cause.details == "string" ? e.cause.details : e.cause?.message ? e.cause.message : e.details;
    })(), u = e.cause instanceof s && e.cause.docsPath || e.docsPath, c = `https://oxlib.sh${u ?? ""}`, l = [
      r || "An error occurred.",
      ...e.metaMessages ? ["", ...e.metaMessages] : [],
      ...a || u ? [
        "",
        a ? `Details: ${a}` : void 0,
        u ? `See: ${c}` : void 0
      ] : []
    ].filter((d) => typeof d == "string").join(`
`);
    super(l, e.cause ? { cause: e.cause } : void 0), Object.defineProperty(this, "details", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "docs", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "docsPath", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "shortMessage", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "cause", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "BaseError"
    }), Object.defineProperty(this, "version", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: `ox@${n()}`
    }), this.cause = e.cause, this.details = a, this.docs = c, this.docsPath = u, this.shortMessage = r;
  }
  walk(r) {
    return i(this, r);
  }
}
function i(t, r) {
  return r?.(t) ? t : t && typeof t == "object" && "cause" in t && t.cause ? i(t.cause, r) : r ? null : t;
}
export {
  s as BaseError
};
