// @coinbase/cdp-core/dist/esm/index.native42.js
import { InputValidationError as a } from "./index.native24.js";
import { getPlatformCrypto as l } from "./index.native6.js";
function y(e, o) {
  if (e.period !== void 0 && e.periodInDays !== void 0)
    throw new a(
      "Cannot specify both 'period' and 'periodInDays'. Please provide only one."
    );
  if (e.period === void 0 && e.periodInDays === void 0)
    throw new a("Must specify either 'period' (in seconds) or 'periodInDays'.");
  const t = e.period ?? e.periodInDays * 24 * 60 * 60, d = /* @__PURE__ */ new Date(), c = e.start ?? d, r = e.end, f = Math.floor(c.getTime() / 1e3), E = r ? Math.floor(r.getTime() / 1e3) : 281474976710655;
  return {
    account: e.account,
    spender: e.spender,
    allowance: e.allowance,
    token: D(e.token, o),
    period: t,
    start: f,
    end: E,
    salt: e.salt ?? l().generateRandomSalt(),
    extraData: e.extraData ?? "0x"
  };
}
function D(e, o) {
  if (e === "eth")
    return "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
  if (e === "usdc") {
    const t = n[o]?.usdc;
    if (!t)
      throw new a(
        `Automatic token address lookup for ${e} is not supported on ${o}. Please provide the token address manually.`
      );
    return t;
  }
  return e;
}
const n = {
  base: {
    usdc: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913"
  },
  "base-sepolia": {
    usdc: "0x036CbD53842c5426634e7929541eC2318f3dCF7e"
  }
};
export {
  y as toSpendPermission
};
