// @coinbase/cdp-core/dist/esm/index.native138.js
function n() {
  try {
    return typeof navigator < "u" && navigator.product === "ReactNative";
  } catch {
    return !1;
  }
}
function r() {
  try {
    if (typeof window > "u")
      return typeof process < "u" && process.env && "__NEXT_PROCESSED_ENV" in process.env;
    const e = window;
    return !!(e.__NEXT_DATA__ || e.next || document.getElementById("__next"));
  } catch {
    return !1;
  }
}
function o() {
  try {
    if (typeof window > "u")
      return !1;
    const e = window;
    return !!(e.__VUE__ || e.__VUE_DEVTOOLS_GLOBAL_HOOK__ || document.querySelector("[data-v-]"));
  } catch {
    return !1;
  }
}
function u() {
  try {
    if (typeof window > "u")
      return !1;
    const e = window;
    return e.__REACT_DEVTOOLS_GLOBAL_HOOK__ || e.React ? !0 : !!document.querySelector(
      "[data-reactroot], [data-reactid], ._reactRoot, ._reactRootContainer"
    );
  } catch {
    return !1;
  }
}
function a() {
  return u() ? "react" : n() ? "react-native" : r() ? "nextjs" : o() ? "vue" : "unknown";
}
let t = null;
function i() {
  return t === null && (t = a()), t;
}
export {
  a as detectFramework,
  i as getFramework
};
