// @coinbase/cdp-api-client/dist/esm/index5.js
import { cdpApiClient as s } from "./index7.js";
const d = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/init`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), i = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/email`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), c = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/sms`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), p = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/verify/oauth/${t}`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), l = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/refresh`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), u = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/logout`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), m = (e, t, n) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/custom`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: t
  },
  n
), h = (e, t) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/auth/custom/authenticate`,
    method: "POST"
  },
  t
), T = (e, t, n) => s(
  { url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}`, method: "GET" },
  n
), v = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/wallet-secrets`,
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), $ = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), j = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), S = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), y = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/send/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), E = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/message`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), w = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/sign/typed-data`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), C = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/export`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), b = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), O = (e, t, n, a, o) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/user-operations/${a}`,
    method: "GET"
  },
  o
), P = (e, t, n, a, o) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/send`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  o
), A = (e, t, n, a, o) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  o
), U = (e, t, n, a, o) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions/list`,
    method: "GET",
    params: a
  },
  o
), g = (e, t, n, a, o) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/evm/smart-accounts/${n}/spend-permissions/revoke`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: a
  },
  o
), W = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), f = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/export`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), x = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), k = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign/message`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), G = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/sign/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
), H = (e, t, n, a) => s(
  {
    url: `/v2/embedded-wallet-api/projects/${e}/end-users/${t}/solana/send/transaction`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: n
  },
  a
);
export {
  h as authenticateUserWithCustomAuth,
  m as createCustomAuthConfig,
  $ as createEndUserEvmAccount,
  b as createEndUserEvmSmartAccount,
  W as createEndUserSolanaAccount,
  A as createSpendPermissionWithEndUserAccount,
  C as exportEndUserEvmAccount,
  f as exportEndUserSolanaAccount,
  T as getAuthenticatedEndUser,
  O as getUserOperationWithEndUserAccount,
  d as initiateAuthentication,
  U as listSpendPermissionsWithEndUserAccount,
  u as logOutEndUser,
  l as refreshAccessToken,
  v as registerTemporaryWalletSecret,
  g as revokeSpendPermissionWithEndUserAccount,
  y as sendEvmTransactionWithEndUserAccount,
  H as sendSolanaTransactionWithEndUserAccount,
  P as sendUserOperationWithEndUserAccount,
  j as signEvmHashWithEndUserAccount,
  E as signEvmMessageWithEndUserAccount,
  S as signEvmTransactionWithEndUserAccount,
  w as signEvmTypedDataWithEndUserAccount,
  x as signSolanaHashWithEndUserAccount,
  k as signSolanaMessageWithEndUserAccount,
  G as signSolanaTransactionWithEndUserAccount,
  i as verifyEmailAuthentication,
  p as verifyOAuthCode,
  c as verifySmsAuthentication
};
