// @coinbase/cdp-api-client/dist/esm/index8.js
const u = {
  unexpected_error: "unexpected_error",
  unauthorized: "unauthorized",
  not_found: "not_found",
  bad_gateway: "bad_gateway",
  service_unavailable: "service_unavailable",
  unknown: "unknown"
};
class a extends Error {
  statusCode;
  errorType;
  errorMessage;
  correlationId;
  errorLink;
  /**
   * Constructor for the APIError class
   *
   * @param statusCode - The HTTP status code
   * @param errorType - The type of error
   * @param errorMessage - The error message
   * @param correlationId - The correlation ID
   * @param errorLink - URL to documentation about this error
   * @param cause - The cause of the error
   */
  constructor(e, o, t, n, s, i) {
    super(t, { cause: i }), this.name = "APIError", this.statusCode = e, this.errorType = o, this.errorMessage = t, n !== void 0 && (this.correlationId = n), s !== void 0 && (this.errorLink = s);
  }
  /**
   * Convert the error to a JSON object, excluding undefined properties
   *
   * @returns The error as a JSON object
   */
  toJSON() {
    return {
      name: this.name,
      statusCode: this.statusCode,
      errorType: this.errorType,
      errorMessage: this.errorMessage,
      ...this.correlationId && { correlationId: this.correlationId },
      ...this.errorLink && { errorLink: this.errorLink }
    };
  }
}
class d extends a {
  /**
   * Constructor for the UnknownApiError class
   *
   * @param errorType - The type of error
   * @param errorMessage - The error message
   * @param cause - The cause of the error
   */
  constructor(e, o, t) {
    super(0, e, o, void 0, void 0, t), this.name = "UnknownApiError";
  }
}
class c extends Error {
  /**
   * Constructor for the UnknownError class
   *
   * @param message - The error message
   * @param cause - The cause of the error
   */
  constructor(e, o) {
    super(e, { cause: o }), this.name = "UnknownError";
  }
}
function p(r) {
  return r !== null && typeof r == "object" && "errorType" in r && typeof r.errorType == "string" && "errorMessage" in r && typeof r.errorMessage == "string";
}
export {
  a as APIError,
  u as HttpErrorType,
  d as UnknownApiError,
  c as UnknownError,
  p as isOpenAPIError
};
