// @coinbase/cdp-api-client/dist/esm/index.js
import { Analytics as e } from "./index2.js";
import "./index3.js";
import { VERSION as t } from "./index4.js";
import { authenticateUserWithCustomAuth as s, createCustomAuthConfig as p, createEndUserEvmAccount as c, createEndUserEvmSmartAccount as d, createEndUserSolanaAccount as m, createSpendPermissionWithEndUserAccount as u, exportEndUserEvmAccount as l, exportEndUserSolanaAccount as y, getAuthenticatedEndUser as S, getUserOperationWithEndUserAccount as T, initiateAuthentication as E, listSpendPermissionsWithEndUserAccount as A, logOutEndUser as h, refreshAccessToken as C, registerTemporaryWalletSecret as O, revokeSpendPermissionWithEndUserAccount as v, sendEvmTransactionWithEndUserAccount as g, sendSolanaTransactionWithEndUserAccount as R, sendUserOperationWithEndUserAccount as U, signEvmHashWithEndUserAccount as k, signEvmMessageWithEndUserAccount as P, signEvmTransactionWithEndUserAccount as N, signEvmTypedDataWithEndUserAccount as w, signSolanaHashWithEndUserAccount as f, signSolanaMessageWithEndUserAccount as I, signSolanaTransactionWithEndUserAccount as W, verifyEmailAuthentication as D, verifyOAuthCode as M, verifySmsAuthentication as V } from "./index5.js";
import { AbiFunctionType as L, AbiInputType as X, AbiStateMutability as q, AccountType as x, AchDetailsAccountType as F, AddressRiskNetwork as B, AddressRiskRiskLevel as H, AssetType as K, CapabilityStatus as j, CoolPaymentMethodType as G, CreatePaymentMethodRequestAchType as J, CreatePaymentMethodRequestFedwireType as Q, CustomerType as _, DeveloperJWTAuthenticationType as z, EmailAuthenticationType as Y, ErrorType as Z, EthValueCriterionOperator as $, EthValueCriterionType as ee, EvmAddressCriterionOperator as te, EvmAddressCriterionType as re, EvmDataCriterionType as ne, EvmDataParameterConditionListOperator as ie, EvmDataParameterConditionOperator as oe, EvmMessageCriterionType as ae, EvmNetworkCriterionNetworksItem as se, EvmNetworkCriterionOperator as pe, EvmNetworkCriterionType as ce, EvmSwapsNetwork as de, EvmTypedAddressConditionOperator as me, EvmTypedNumericalConditionOperator as ue, EvmUserOperationNetwork as le, EvmUserOperationStatus as ye, InitiateEmailAuthenticationNextStepType as Se, InitiateEmailAuthenticationRequestType as Te, InitiateSmsAuthenticationNextStepType as Ee, InitiateSmsAuthenticationRequestType as Ae, InvitationStatus as he, KnownAbiType as Ce, KnownIdlType as Oe, LimitPeriod as ve, LinkAccountType as ge, LinkableEntityRole as Re, LinkableEntityType as Ue, ListEvmTokenBalancesNetwork as ke, ListSolanaTokenBalancesNetwork as Pe, MintAddressCriterionOperator as Ne, MintAddressCriterionType as we, NetUSDChangeCriterionOperator as fe, NetUSDChangeCriterionType as Ie, Network as We, OAuth2ProviderType as De, OnchainDataResultSchemaColumnsItemType as Me, OnrampOrderFeeType as Ve, OnrampOrderPaymentMethodTypeId as be, OnrampOrderStatus as Le, OnrampPaymentLinkType as Xe, OnrampQuotePaymentMethodTypeId as qe, PermissionCatalogObjectsItemType as xe, PermissionSubjectType as Fe, PermissionVerb as Be, PolicyScope as He, PrepareUserOperationRuleAction as Ke, PrepareUserOperationRuleOperation as je, ProgramIdCriterionOperator as Ge, ProgramIdCriterionType as Je, RefreshAccessTokenBodyGrantType as Qe, RequirementErrorCode as _e, RequirementField as ze, SendEvmTransactionRuleAction as Ye, SendEvmTransactionRuleOperation as Ze, SendEvmTransactionWithEndUserAccountBodyNetwork as $e, SendSolTransactionRuleAction as et, SendSolTransactionRuleOperation as tt, SendSolanaTransactionWithEndUserAccountBodyNetwork as rt, SendUserOperationRuleAction as nt, SendUserOperationRuleOperation as it, SignEvmHashRuleAction as ot, SignEvmHashRuleOperation as at, SignEvmMessageRuleAction as st, SignEvmMessageRuleOperation as pt, SignEvmTransactionRuleAction as ct, SignEvmTransactionRuleOperation as dt, SignEvmTypedDataFieldCriterionType as mt, SignEvmTypedDataRuleAction as ut, SignEvmTypedDataRuleOperation as lt, SignEvmTypedDataVerifyingContractCriterionOperator as yt, SignEvmTypedDataVerifyingContractCriterionType as St, SignSolMessageRuleAction as Tt, SignSolMessageRuleOperation as Et, SignSolTransactionRuleAction as At, SignSolTransactionRuleOperation as ht, SmsAuthenticationType as Ct, SolAddressCriterionOperator as Ot, SolAddressCriterionType as vt, SolDataCriterionType as gt, SolDataParameterConditionListOperator as Rt, SolDataParameterConditionOperator as Ut, SolMessageCriterionType as kt, SolNetworkCriterionNetworksItem as Pt, SolNetworkCriterionOperator as Nt, SolNetworkCriterionType as wt, SolValueCriterionOperator as ft, SolValueCriterionType as It, SpendPermissionNetwork as Wt, SplAddressCriterionOperator as Dt, SplAddressCriterionType as Mt, SplValueCriterionOperator as Vt, SplValueCriterionType as bt, TransferCancellationReason as Lt, TransferFeeName as Xt, TransferRequestAmountType as qt, TransferReversalReason as xt, TransferStatus as Ft, X402DiscoveryResourceType as Bt, X402PaymentPayloadNetwork as Ht, X402PaymentPayloadScheme as Kt, X402PaymentRequirementsNetwork as jt, X402PaymentRequirementsScheme as Gt, X402SettleErrorReason as Jt, X402SupportedPaymentKindNetwork as Qt, X402SupportedPaymentKindScheme as _t, X402VerifyInvalidReason as zt, X402Version as Yt } from "./index6.js";
import { configureCdpApiClient as $t, setAuthManager as er } from "./index7.js";
import { APIError as rr, HttpErrorType as nr, UnknownApiError as ir, UnknownError as or, isOpenAPIError as ar } from "./index8.js";
e.registerPackageVersion("api_client", t);
export {
  rr as APIError,
  L as AbiFunctionType,
  X as AbiInputType,
  q as AbiStateMutability,
  x as AccountType,
  F as AchDetailsAccountType,
  B as AddressRiskNetwork,
  H as AddressRiskRiskLevel,
  K as AssetType,
  j as CapabilityStatus,
  G as CoolPaymentMethodType,
  J as CreatePaymentMethodRequestAchType,
  Q as CreatePaymentMethodRequestFedwireType,
  _ as CustomerType,
  z as DeveloperJWTAuthenticationType,
  Y as EmailAuthenticationType,
  Z as ErrorType,
  $ as EthValueCriterionOperator,
  ee as EthValueCriterionType,
  te as EvmAddressCriterionOperator,
  re as EvmAddressCriterionType,
  ne as EvmDataCriterionType,
  ie as EvmDataParameterConditionListOperator,
  oe as EvmDataParameterConditionOperator,
  ae as EvmMessageCriterionType,
  se as EvmNetworkCriterionNetworksItem,
  pe as EvmNetworkCriterionOperator,
  ce as EvmNetworkCriterionType,
  de as EvmSwapsNetwork,
  me as EvmTypedAddressConditionOperator,
  ue as EvmTypedNumericalConditionOperator,
  le as EvmUserOperationNetwork,
  ye as EvmUserOperationStatus,
  nr as HttpErrorType,
  Se as InitiateEmailAuthenticationNextStepType,
  Te as InitiateEmailAuthenticationRequestType,
  Ee as InitiateSmsAuthenticationNextStepType,
  Ae as InitiateSmsAuthenticationRequestType,
  he as InvitationStatus,
  Ce as KnownAbiType,
  Oe as KnownIdlType,
  ve as LimitPeriod,
  ge as LinkAccountType,
  Re as LinkableEntityRole,
  Ue as LinkableEntityType,
  ke as ListEvmTokenBalancesNetwork,
  Pe as ListSolanaTokenBalancesNetwork,
  Ne as MintAddressCriterionOperator,
  we as MintAddressCriterionType,
  fe as NetUSDChangeCriterionOperator,
  Ie as NetUSDChangeCriterionType,
  We as Network,
  De as OAuth2ProviderType,
  Me as OnchainDataResultSchemaColumnsItemType,
  Ve as OnrampOrderFeeType,
  be as OnrampOrderPaymentMethodTypeId,
  Le as OnrampOrderStatus,
  Xe as OnrampPaymentLinkType,
  qe as OnrampQuotePaymentMethodTypeId,
  xe as PermissionCatalogObjectsItemType,
  Fe as PermissionSubjectType,
  Be as PermissionVerb,
  He as PolicyScope,
  Ke as PrepareUserOperationRuleAction,
  je as PrepareUserOperationRuleOperation,
  Ge as ProgramIdCriterionOperator,
  Je as ProgramIdCriterionType,
  Qe as RefreshAccessTokenBodyGrantType,
  _e as RequirementErrorCode,
  ze as RequirementField,
  Ye as SendEvmTransactionRuleAction,
  Ze as SendEvmTransactionRuleOperation,
  $e as SendEvmTransactionWithEndUserAccountBodyNetwork,
  et as SendSolTransactionRuleAction,
  tt as SendSolTransactionRuleOperation,
  rt as SendSolanaTransactionWithEndUserAccountBodyNetwork,
  nt as SendUserOperationRuleAction,
  it as SendUserOperationRuleOperation,
  ot as SignEvmHashRuleAction,
  at as SignEvmHashRuleOperation,
  st as SignEvmMessageRuleAction,
  pt as SignEvmMessageRuleOperation,
  ct as SignEvmTransactionRuleAction,
  dt as SignEvmTransactionRuleOperation,
  mt as SignEvmTypedDataFieldCriterionType,
  ut as SignEvmTypedDataRuleAction,
  lt as SignEvmTypedDataRuleOperation,
  yt as SignEvmTypedDataVerifyingContractCriterionOperator,
  St as SignEvmTypedDataVerifyingContractCriterionType,
  Tt as SignSolMessageRuleAction,
  Et as SignSolMessageRuleOperation,
  At as SignSolTransactionRuleAction,
  ht as SignSolTransactionRuleOperation,
  Ct as SmsAuthenticationType,
  Ot as SolAddressCriterionOperator,
  vt as SolAddressCriterionType,
  gt as SolDataCriterionType,
  Rt as SolDataParameterConditionListOperator,
  Ut as SolDataParameterConditionOperator,
  kt as SolMessageCriterionType,
  Pt as SolNetworkCriterionNetworksItem,
  Nt as SolNetworkCriterionOperator,
  wt as SolNetworkCriterionType,
  ft as SolValueCriterionOperator,
  It as SolValueCriterionType,
  Wt as SpendPermissionNetwork,
  Dt as SplAddressCriterionOperator,
  Mt as SplAddressCriterionType,
  Vt as SplValueCriterionOperator,
  bt as SplValueCriterionType,
  Lt as TransferCancellationReason,
  Xt as TransferFeeName,
  qt as TransferRequestAmountType,
  xt as TransferReversalReason,
  Ft as TransferStatus,
  ir as UnknownApiError,
  or as UnknownError,
  Bt as X402DiscoveryResourceType,
  Ht as X402PaymentPayloadNetwork,
  Kt as X402PaymentPayloadScheme,
  jt as X402PaymentRequirementsNetwork,
  Gt as X402PaymentRequirementsScheme,
  Jt as X402SettleErrorReason,
  Qt as X402SupportedPaymentKindNetwork,
  _t as X402SupportedPaymentKindScheme,
  zt as X402VerifyInvalidReason,
  Yt as X402Version,
  s as authenticateUserWithCustomAuth,
  $t as configureCdpApiClient,
  p as createCustomAuthConfig,
  c as createEndUserEvmAccount,
  d as createEndUserEvmSmartAccount,
  m as createEndUserSolanaAccount,
  u as createSpendPermissionWithEndUserAccount,
  l as exportEndUserEvmAccount,
  y as exportEndUserSolanaAccount,
  S as getAuthenticatedEndUser,
  T as getUserOperationWithEndUserAccount,
  E as initiateAuthentication,
  ar as isOpenAPIError,
  A as listSpendPermissionsWithEndUserAccount,
  h as logOutEndUser,
  C as refreshAccessToken,
  O as registerTemporaryWalletSecret,
  v as revokeSpendPermissionWithEndUserAccount,
  g as sendEvmTransactionWithEndUserAccount,
  R as sendSolanaTransactionWithEndUserAccount,
  U as sendUserOperationWithEndUserAccount,
  er as setAuthManager,
  k as signEvmHashWithEndUserAccount,
  P as signEvmMessageWithEndUserAccount,
  N as signEvmTransactionWithEndUserAccount,
  w as signEvmTypedDataWithEndUserAccount,
  f as signSolanaHashWithEndUserAccount,
  I as signSolanaMessageWithEndUserAccount,
  W as signSolanaTransactionWithEndUserAccount,
  D as verifyEmailAuthentication,
  M as verifyOAuthCode,
  V as verifySmsAuthentication
};
