// @coinbase/cdp-core/dist/native/index.native62.js
import { handleAccounts as t } from "./index.native114.js";
import { handleEthSign as e } from "./index.native115.js";
import { handleGetCallsStatus as a } from "./index.native116.js";
import { handleGetCapabilities as n } from "./index.native117.js";
import { handlePersonalSign as o } from "./index.native118.js";
import { handleRequestAccounts as r } from "./index.native119.js";
import { handleSendCalls as i } from "./index.native120.js";
import { handleSendTransaction as l } from "./index.native121.js";
import { handleSignTypedData as h } from "./index.native122.js";
import { handleChainId as s } from "./index.native123.js";
import { handleDisconnect as m } from "./index.native124.js";
import { handleSwitchEthereumChain as d } from "./index.native125.js";
function D() {
  return {
    eth_accounts: t,
    eth_requestAccounts: r,
    personal_sign: o,
    eth_sendTransaction: l,
    eth_signTypedData_v4: h,
    wallet_disconnect: m,
    wallet_switchEthereumChain: d,
    eth_sign: e,
    eth_chainId: s,
    wallet_sendCalls: i,
    wallet_getCallsStatus: a,
    wallet_getCapabilities: n
  };
}
export {
  D as createEOAHandlers
};
