// @coinbase/cdp-core/dist/native/index.native3.js
import * as u from "expo-web-browser";
import { WebBrowserResultType as o } from "expo-web-browser";
import { getCoreOAuthManager as s, getConfig as l } from "./index.native28.js";
const d = {
  getRedirectUrl: () => {
    const e = l().nativeOAuthCallback;
    if (!e)
      throw new Error("nativeOAuthCallback must be configured in order to use native OAuth");
    return e;
  },
  getOAuthParams: (e) => {
    if (!e)
      return null;
    const r = new URL(e), t = r.searchParams.get("code") ?? void 0, a = r.searchParams.get("provider_type") ?? void 0, i = r.searchParams.get("flow_id") ?? void 0, c = r.searchParams.get("error") ?? void 0, n = r.searchParams.get("error_description") ?? void 0;
    return (!t || !a || !i) && !c && !n ? null : { code: t, providerType: a, flowId: i, error: c, errorDescription: n };
  },
  cleanupOAuthParams: () => {
  },
  initiateOAuthFlow: async (e) => {
    const r = d.getRedirectUrl(), t = await u.openAuthSessionAsync(e, r);
    switch (t.type) {
      case "success":
        if (!t.url)
          throw new Error("URL is required for success result");
        s().handleOAuthCode(t.url);
        break;
      case o.CANCEL:
        s().setOAuthFlowState({
          status: "error",
          errorDescription: "User cancelled the OAuth flow"
        });
        break;
      case o.DISMISS:
        s().setOAuthFlowState({
          status: "error",
          errorDescription: "User dismissed the OAuth flow"
        });
        break;
      case o.LOCKED:
        s().setOAuthFlowState({
          status: "error",
          errorDescription: "The browser was locked"
        });
        break;
      case o.OPENED:
        break;
      default:
        throw new Error(
          `Unexpected WebBrowser result type: ${t.type}`
        );
    }
  }
};
export {
  d as nativeOAuth
};
