// @coinbase/cdp-core/dist/native/index.native155.js
import { BaseError as f } from "./index.native65.js";
import { pad as g } from "./index.native174.js";
function m(t, i = {}) {
  const { signed: r, size: e } = i, o = BigInt(t);
  let n;
  e ? r ? n = (1n << BigInt(e) * 8n - 1n) - 1n : n = 2n ** (BigInt(e) * 8n) - 1n : typeof t == "number" && (n = BigInt(Number.MAX_SAFE_INTEGER));
  const u = typeof n == "bigint" && r ? -n - 1n : 0;
  if (n && o > n || o < u) {
    const s = typeof t == "bigint" ? "n" : "";
    throw new d({
      max: n ? `${n}${s}` : void 0,
      min: `${u}${s}`,
      signed: r,
      size: e,
      value: `${t}${s}`
    });
  }
  const a = `0x${(r && o < 0 ? (1n << BigInt(e * 8)) + BigInt(o) : o).toString(16)}`;
  return e ? c(a, e) : a;
}
function c(t, i) {
  return g(t, { dir: "left", size: i });
}
class d extends f {
  constructor({ max: i, min: r, signed: e, size: o, value: n }) {
    super(`Number \`${n}\` is not in safe${o ? ` ${o * 8}-bit` : ""}${e ? " signed" : " unsigned"} integer range ${i ? `(\`${r}\` to \`${i}\`)` : `(above \`${r}\`)`}`), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Hex.IntegerOutOfRangeError"
    });
  }
}
class p extends f {
  constructor({ size: i, targetSize: r, type: e }) {
    super(`${e.charAt(0).toUpperCase()}${e.slice(1).toLowerCase()} size (\`${i}\`) exceeds padding size (\`${r}\`).`), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: "Hex.SizeExceedsPaddingSizeError"
    });
  }
}
export {
  d as IntegerOutOfRangeError,
  p as SizeExceedsPaddingSizeError,
  m as fromNumber,
  c as padLeft
};
