// @coinbase/cdp-core/dist/native/index.native33.js
import { AuthManager as a } from "./index.native21.js";
import { getConfig as o } from "./index.native28.js";
import { CustomAuthError as s } from "./index.native22.js";
const n = 60 * 60 * 1e3;
class l extends a {
  /**
   * Signs out the user, clearing all authentication state.
   * For custom auth, we only clear local state without calling the CDP logout API,
   * as authentication is managed by the external identity provider.
   */
  async signOut() {
    await this.clearAuthState();
  }
  /**
   * Gets the access token by calling the customAuth.getJwt callback.
   * Returns null if the user is not signed in.
   *
   * @returns The access token from the custom auth provider.
   */
  async getToken() {
    const e = o();
    if (!e.customAuth)
      return console.error("Custom auth configuration not found"), await this.clearAuthState(), null;
    try {
      const t = await e.customAuth.getJwt();
      return t ? (this.authState && (this.authState.accessToken = t, this.authState.expiresAt = Date.now() + n), t) : (await this.clearAuthState(), null);
    } catch (t) {
      const r = new s(
        t instanceof Error ? t.message : "Failed to get custom auth JWT"
      );
      return console.error(r), await this.clearAuthState(), null;
    }
  }
  /**
   * Internal async initialization logic for custom auth.
   * No-op for custom auth since tokens are provided via customAuth.getJwt callback.
   * Unlike standard CDP auth, custom auth doesn't need token refresh or OAuth handling.
   */
  async _doInitialize() {
  }
  /**
   * Custom auth always fetches fresh tokens, so never needs to "refresh" in the traditional sense.
   *
   * @returns Always false since custom auth doesn't use token refresh logic.
   */
  shouldRefreshToken() {
    return !1;
  }
  /**
   * Custom auth doesn't use CDP's token refresh mechanism.
   * Tokens are always provided fresh from the customAuth.getJwt callback.
   * This method should never be called, but we override it as a defensive measure.
   */
  async refreshAccessToken() {
  }
  /**
   * Custom auth provides fresh tokens on each request via customAuth.getJwt,
   * so no scheduled refresh is needed. This method is a no-op for custom auth.
   */
  scheduleTokenRefresh() {
    this.cancelTokenRefresh();
  }
}
export {
  n as CUSTOM_AUTH_REFRESH_BUFFER_MS,
  l as CustomAuthManager
};
