// @coinbase/cdp-core/dist/native/index.native.js
import { nativeCrypto as r } from "./index.native2.js";
import { nativeOAuth as o } from "./index.native3.js";
import { nativeSecureStorage as e } from "./index.native4.js";
import { nativeStorage as t } from "./index.native5.js";
import { setPlatformServices as n } from "./index.native6.js";
import { authenticateWithJWT as g, exportEvmAccount as l, exportSolanaAccount as S, getAccessToken as x, getAccessTokenExpiration as A, getCurrentUser as d, getUserOperation as h, initialize as v, isSignedIn as T, onAuthStateChange as O, onOAuthStateChange as y, sendEvmTransaction as P, sendSolanaTransaction as k, sendUserOperation as W, signEvmHash as C, signEvmMessage as I, signEvmTransaction as U, signEvmTypedData as D, signInWithEmail as R, signInWithOAuth as M, signInWithSms as N, signOut as V, signSolanaMessage as w, signSolanaTransaction as B, verifyEmailOTP as H, verifyOAuth as _, verifySmsOTP as b } from "./index.native7.js";
import { createSpendPermission as G } from "./index.native8.js";
import { listSpendPermissions as X } from "./index.native9.js";
import { APIError as q, ErrorType as F, HttpErrorType as K, OAuth2ProviderType as L, SendEvmTransactionWithEndUserAccountBodyNetwork as Q, SendSolanaTransactionWithEndUserAccountBodyNetwork as Y } from "@coinbase/cdp-api-client";
import { revokeSpendPermission as $ } from "./index.native10.js";
import { getCurrentUserSync as or } from "./index.native11.js";
import { fetchWithX402 as tr } from "./index.native12.js";
import { linkGoogle as ir } from "./index.native13.js";
import { linkSms as mr } from "./index.native14.js";
import { linkEmail as pr } from "./index.native15.js";
import { linkOAuth as fr } from "./index.native16.js";
import { linkApple as ur } from "./index.native17.js";
import { createEvmEoaAccount as lr } from "./index.native18.js";
import { createEvmSmartAccount as xr } from "./index.native19.js";
import { createSolanaAccount as dr } from "./index.native20.js";
import { AuthManager as vr } from "./index.native21.js";
import { CustomAuthError as Or } from "./index.native22.js";
import { toViemAccount as Pr } from "./index.native23.js";
import { InputValidationError as Wr, InternalError as Cr, ModuleResolutionError as Ir, ValidationError as Ur } from "./index.native24.js";
import { createCDPEmbeddedWallet as Rr } from "./index.native25.js";
import { EIP1193ProviderError as Nr, STANDARD_ERROR_CODES as Vr } from "./index.native26.js";
import "viem";
import "zustand";
n({
  crypto: r,
  storage: t,
  secureStorage: e,
  oauth: o
});
export {
  q as APIError,
  vr as AuthManager,
  Or as CustomAuthError,
  Nr as EIP1193ProviderError,
  F as ErrorType,
  K as HttpErrorType,
  Wr as InputValidationError,
  Cr as InternalError,
  Ir as ModuleResolutionError,
  L as OAuth2ProviderType,
  Vr as STANDARD_ERROR_CODES,
  Q as SendEvmTransactionWithEndUserAccountBodyNetwork,
  Y as SendSolanaTransactionWithEndUserAccountBodyNetwork,
  Ur as ValidationError,
  g as authenticateWithJWT,
  Rr as createCDPEmbeddedWallet,
  lr as createEvmEoaAccount,
  xr as createEvmSmartAccount,
  dr as createSolanaAccount,
  G as createSpendPermission,
  l as exportEvmAccount,
  S as exportSolanaAccount,
  tr as fetchWithX402,
  x as getAccessToken,
  A as getAccessTokenExpiration,
  d as getCurrentUser,
  or as getCurrentUserSync,
  h as getUserOperation,
  v as initialize,
  T as isSignedIn,
  ur as linkApple,
  pr as linkEmail,
  ir as linkGoogle,
  fr as linkOAuth,
  mr as linkSms,
  X as listSpendPermissions,
  O as onAuthStateChange,
  y as onOAuthStateChange,
  $ as revokeSpendPermission,
  P as sendEvmTransaction,
  k as sendSolanaTransaction,
  W as sendUserOperation,
  C as signEvmHash,
  I as signEvmMessage,
  U as signEvmTransaction,
  D as signEvmTypedData,
  R as signInWithEmail,
  M as signInWithOAuth,
  N as signInWithSms,
  V as signOut,
  w as signSolanaMessage,
  B as signSolanaTransaction,
  Pr as toViemAccount,
  H as verifyEmailOTP,
  _ as verifyOAuth,
  b as verifySmsOTP
};
