// @coinbase/cdp-core/dist/native/index.native9.js
import { listSpendPermissionsWithEndUserAccount as a } from "@coinbase/cdp-api-client";
import { Analytics as i } from "./index.native30.js";
import "./index.native31.js";
import { withAuth as o } from "./index.native36.js";
import { getConfig as r, getCoreAuthManager as m } from "./index.native28.js";
const k = async (e) => (i.sendActionCallEvent({ name: "list_spend_permissions" }), r().useMock ? {
  spendPermissions: [
    {
      permission: {
        account: e.evmSmartAccount,
        spender: "0x5678901234567890123456789012345678901234",
        token: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        allowance: "1000000000000000000",
        period: "86400",
        start: "1640995200",
        end: "1672531200",
        salt: "123456789",
        extraData: "0x"
      },
      permissionHash: "0x1234567890123456789012345678901234567890123456789012345678901234",
      revoked: !1,
      createdAt: "2024-01-01T00:00:00Z",
      network: e.network
    }
  ],
  hasNextPage: !1
} : o(e, m(), async ({ user: t }) => {
  const n = {};
  e.pageSize !== void 0 && (n.pageSize = e.pageSize), e.pageToken !== void 0 && (n.pageToken = e.pageToken);
  const s = await a(
    r().projectId,
    t.userId,
    e.evmSmartAccount,
    n
  );
  return {
    spendPermissions: s.spendPermissions,
    hasNextPage: !!s.nextPageToken,
    nextPageToken: s.nextPageToken
  };
}));
export {
  k as listSpendPermissions
};
