// @coinbase/cdp-core/dist/native/index.native40.js
import { SendEvmTransactionWithEndUserAccountBodyNetwork as i } from "@coinbase/cdp-api-client";
const t = (e) => !e || typeof e != "object" ? e : Array.isArray(e) ? e.map(t) : Object.keys(e).sort().reduce(
  (r, n) => (r[n] = t(e[n]), r),
  {}
), p = (e) => Object.values(i).includes(
  e
), s = {
  "base-sepolia": 84532,
  base: 8453,
  ethereum: 1,
  "ethereum-sepolia": 11155111,
  avalanche: 43114,
  polygon: 137,
  optimism: 10,
  arbitrum: 42161
}, c = Object.fromEntries(
  Object.entries(s).map(([e, r]) => [r, e])
), u = (e) => Object.values(s).includes(
  e
), a = {
  "base-sepolia": 84532,
  base: 8453,
  ethereum: 1,
  "ethereum-sepolia": 11155111,
  avalanche: 43114,
  polygon: 137,
  optimism: 10,
  arbitrum: 42161,
  zora: 7777777,
  bnb: 56
}, m = Object.fromEntries(
  Object.entries(a).map(([e, r]) => [r, e])
), d = (e) => Object.values(a).includes(
  e
);
export {
  u as isChainIdSupportedForCDPSends,
  d as isChainIdSupportedForCDPUserOps,
  p as isChainSupportedForCDPSends,
  c as sendTransactionChainIdToNameMapping,
  t as sortKeys,
  m as userOperationChainIdToNameMapping,
  a as userOperationNameToChainIdMapping
};
