// @coinbase/cdp-core/dist/native/index.native25.js
import { createStore as c } from "zustand";
import { subscribeWithSelector as p, persist as d, createJSONStorage as h } from "./index.native49.js";
import { EventEmitter as f } from "./index.native50.js";
import "./index.native51.js";
import "viem";
import "@coinbase/cdp-api-client";
import { getPlatformStorage as u } from "./index.native6.js";
import { createProvider as g } from "./index.native52.js";
import { onAuthStateChange as l } from "./index.native7.js";
function U(e) {
  if (e.chains.length === 0)
    throw new Error("At least one chain must be configured");
  if (!e.transports)
    throw new Error("Transports must be provided");
  const o = e.chains, i = e.transports;
  if (o.length === 0)
    throw new Error("No chains found in configuration");
  const m = new f(), n = u(), a = {
    getItem: (t) => n.getItem(t),
    setItem: (t, r) => n.setItem(t, r),
    removeItem: (t) => n.removeItem(t)
  }, s = c()(
    p(
      d(
        (t) => ({
          chainId: o[0].id,
          setChainId: (r) => t({ chainId: r }),
          chains: o,
          user: null,
          setUser: (r) => t({ user: r })
        }),
        {
          name: "cdp-provider-store",
          storage: h(() => a)
        }
      )
    )
  );
  return s.persist.rehydrate(), l((t) => {
    s.getState().setUser(t);
  }), {
    provider: g(m, s, {
      chains: o,
      transports: i
    })
  };
}
export {
  U as createCDPEmbeddedWallet
};
